syntax = "proto2";
import "hvac_status.proto";
import "door_status.proto";
import "window_status.proto";
import "light_status.proto";
import "soc_status.proto";
import "vehicle_status.proto";
import "tyre_status.proto";
import "alarm_signal.proto";
import "dtc_resp.proto";
import "did_resp.proto";
import "position_status.proto";
import "attach_file.proto";
import "heating_status.proto";
import "script_channel.proto";
import "fota_rescue_cfg_rslt.proto";

option java_package = "com.nextev.cvs_proto.protobuf.rvs";
option java_outer_classname = "CmdResultMsg";

message CmdResult {

    optional int64 command_id = 1; // command id from control command
    optional ResultStatus status = 2; //command result of execution
    repeated string fail_reason = 3; //fail reason

    optional HVACStatus hvac_status = 4;
    optional DoorStatus door_status = 5;
    optional WindowStatus window_status = 6;
    optional LightStatus light_status = 7;
    optional SOCStatus soc_status = 8;
    optional VehicleStatus vehicle_status = 9;
    optional TyreStatus tyre_status = 10;
    optional AlarmSignal alarm_signal = 11;
    optional PositionStatus position_status = 12;
    optional SOCStatus.ChargingInfo charging_info = 13; //required when vehicle is charging
    optional DTCStatusResp dtc_status_resp = 14;
    optional DIDStatusResp did_status_resp = 15;
    optional FileInfo file_info = 16;
    optional HeatingStatus heating_status = 17;
    optional ScriptChannel script_channel = 18;
    optional FotaRescueCfgRslt fota_rescue_cfg_rslt = 19;


    enum ResultStatus {
        SUCCESS = 0;
        FAILURE = 1;
        RUNNING = 2;
        ADDITIONAL = 3; //reserved, donot use
        NOT_COMPLETE = 7;
    }

}
